File: 0121. Best Time to Buy and Sell Stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        
        if(prices.Length == 0){
            return 0 ;
        }
        
        int min = prices[0];
        int max = 0;
        
        for(int i = 0 ; i<prices.Length ;i++){
            if(prices[i] < min){
                min = prices[i];
            }else if(prices[i]-min > max){
                max = prices[i] - min;
            }
        }
        
        return max;
        
        
    }
}

File: 0121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        if len(prices) == 0:
            return 0
                
        min = prices[0]
        max = 0
                
        for i in prices:
            if i < min:
                min = i
            elif i - min > max:
                max = i - min
                    
        return max